src/3d/castletransform_physics.inc

```diff
@@ -818,7 +815,24 @@ begin
 end;
 
 destructor TSceneManagerWorld.Destroy;
+var
+  CastleRigidBody: TRigidBody;
+  KraftRigidBody: TKraftRigidBody;
+  NextKraftRigidBody: TKraftRigidBody;
 begin
+  KraftRigidBody := FKraftEngine.RigidBodyFirst;
```

Comment:
Destructor may be called even when some exception occurs in the middle of the constructor. So destructor body must be "robust", i.e. prepared to to handle the case when *anything* can be nil. In this case, it means that you should put the whole block of code here under `if FKraftEngine <> nil then ...`.